versioned/tiered/src/main/java/com/dremio/nessie/versioned/impl/condition/ExpressionPath.java

```diff
@@ -177,7 +182,7 @@ public PathSegment alias(AliasCollector c) {
     public abstract String getName();
 
     @Override
-    public PathSegment alias(AliasCollector c) {
+    public NameSegment alias(AliasCollector c) {
```

Comment:
What does alias() do?
There is one that returns an ExpressionFunction, and one that returns a NameSegment.
Should each function have an unique name? Or do we want those objects to have the same alias() function?